google/cloud/bigquery/exceptions.py

```diff
@@ -18,4 +18,8 @@ class BigQueryError(Exception):
 
 
 class LegacyBigQueryStorageError(BigQueryError):
-    """Raised when too old a version of BigQuery Storage extra is detected at runtime."""
+    """Raised when too old a version of BigQuery Storage extra is detected at runtime.
+
+    .. deprecated:: 2.2.0
+       Not used anymore and will be removed in the future.
```

Comment:
Note to self - this should have been 2.22.0, but it's not obsolete, since we've just made that [release](https://github.com/googleapis/python-bigquery/pull/771#issuecomment-882856588). Needs to be adjusted once we decide what the next version will be.

Reply:
Done. Removed the custom exception class altogether, because we'll do a major version bump, thus a breaking change is acceptable.